### PR TITLE
[#105] Drainer pod uses un-resolvable hostname for it's cluster connection

### LINKED ISSUE #105: Drainer pod uses un-resolvable hostname for it's cluster connection
**Describe the bug**
During scaledown the drainer.sh uses its hostname in its connector for clusterconnection. However as the drainer pod is a standalone pod it's hostname (pod name) cannot be resolved by other broker pods. As a result when other broker pods try to use the hostname to establish a cluster connection to the drainer pod, "UnknownHostException" exception is thrown and as a result the broker keeps retrying (for 20 times as per current configure).
The simple fix is to just use pod's IP which is resolvable across the cluster.

**To Reproduce**
1. Get X version of the source
2. '...'

**Platform (please complete the following information):**
 - OS: [e.g. Fedora]
 - Version [e.g. 32]
 - CEKit version [e.g. 3.5.0]
 - Container build tool [e.g. podman]

**Expected behavior**
Optional, use if not obvious. A clear and concise description of what you expected to happen.

**Screenshots**
Optional, add screenshots to help explain your problem.

**Additional context**
Optional. Add any other context about the problem here.

### PR DESCRIPTION
**Pull request by gaohoward**
Thanks for submitting your Pull Request!

Please make sure your PR meets the following requirements:

- [ ] Pull Request title is properly formatted: `[CLOUD-XYA] Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket
- [ ] Attached commits represent units of work and are properly formatted
- [ ] You have read and agreed to the Developer Certificate of Origin (DCO) (see `CONTRIBUTING.md`)
- [ ] Every commit contains `Signed-off-by: Your Name <yourname@example.com>` - use `git commit -s`

### CODE REVIEW

File: modules/activemq-artemis-launch/added/drain.sh [hunk 1]
```diff
@@ -1,6 +1,8 @@
 #!/usr/bin/env bash
 
-export BROKER_HOST="$(hostname -f)"
+# use pod IP rather than `hostname -f` which will return pod's name
+# that is not resolvable across the cluster
+export BROKER_HOST=${DRAINER_IP}
```
Comment (brusdev) [inline modules/activemq-artemis-launch/added/drain.sh]: A check to test DRAINER_IP is set could help to detect misconfigurations, i.e.
```
if [ -z "${DRAINER_IP}" ]; then
  echo "[drain.sh] DRAINER_IP is not set"
  sleep 30
  exit 1
fi
```

File: modules/activemq-artemis-launch/added/drain.sh [hunk 2]
```diff
@@ -110,8 +112,12 @@ echo "[drain.sh] scale down target is: $SCALE_TO_BROKER"
 connector="<connector name=\"scaledownconnector\">tcp:\/\/${SCALE_TO_BROKER}:61616<\/connector>"
 sed -i "/<\/connectors>/ s/.*/${connector}\n&/" "${instanceDir}/etc/broker.xml"
 
+connector="<connector name=\"artemis\">tcp:\/\/${BROKER_HOST}:61616<\/connector>"
+sed -i "s/<connector name=\"artemis\">.*<\/connector>/${connector}/" "${instanceDir}/etc/broker.xml"
+
 # Remove the acceptors
 #sed -i -ne "/<acceptors>/ {p;   " -e ":a; n; /<\/acceptors>/ {p; b}; ba}; p" ${instanceDir}/etc/broker.xml
+
```
Comment (brusdev) [inline modules/activemq-artemis-launch/added/drain.sh]: Did you add this new line on purpose?

**Comment by brusdev** [comment]
An alternative solution to the IP address is to pass the full hostname as the Operator does for jolokia client, see https://github.com/artemiscloud/activemq-artemis-operator/blob/main/pkg/utils/jolokia_client/jolokia_client.go#L75

**Comment by gaohoward** [comment]
> An alternative solution to the IP address is to pass the full hostname as the Operator does for jolokia client, see https://github.com/artemiscloud/activemq-artemis-operator/blob/main/pkg/utils/jolokia_client/jolokia_client.go#L75

That may not work. The problem is that the drainer pod is a standalone pod and the DNS record for the pod may not be even generated (because it is not part of the headless service any more). To get a valid resolvable record some more configurations need to be added for this pod. So I think the IP address solution is the simple one to go.

File: modules/activemq-artemis-launch/added/drain.sh [hunk 3]
```diff
@@ -1,8 +1,16 @@
 #!/usr/bin/env bash
 
-export BROKER_HOST="$(hostname -f)"
+if [ -z "${DRAINER_IP}" ]; then
```
Comment (brusdev) [inline modules/activemq-artemis-launch/added/drain.sh]: The name `DRAINER_HOST` would work with hostnames and IP addresses. In the case we decide to pass an hostname form the oeprator.

**Review by brusdev — APPROVED** [review]

File: modules/activemq-artemis-launch/added/drain.sh (hunk 3 above)
Comment (gaohoward) [reply modules/activemq-artemis-launch/added/drain.sh]: +1

Comment (gaohoward) [reply modules/activemq-artemis-launch/added/drain.sh]: I'll change it to DRAINER_HOST

**Review by brusdev — APPROVED** [review]

**Pull request merged**